lib/lotus/validations/validation.rb

```diff
@@ -0,0 +1,34 @@
+module Lotus
+  module Validations
+    module Validation
+      attr_reader :attribute_name, :value, :attributes, :errors
+
+      def initialize(attribute_name, attributes, errors)
+        @attribute_name = attribute_name
+        @value = attributes[attribute_name]
+        @attributes = attributes
+        @errors = errors
+      end
+
+      def call
+        fail NotImplementedError
+      end
+
+      def validation_name
+        @validation_name ||= begin
+                               class_name = self.class.name.split('::').last
```

Comment:
Why passing the attributes hash and not the entity (or any ruby object which responds to _attribute_name_ ?